stix2slider/convert_pattern.py

```diff
@@ -606,53 +609,19 @@ def add_file_pdf_extension_pattern(file_obj, properties, rhs, op, id20):
                             op,
                             PDF_DOCUMENT_INFORMATION_DICT_MAP,
                             id20)
-        else:
-            warn("%s is not a legal property in the pattern of %s", 303, prop_name1, id20)
-
-
-# _PE_BINARY_FILE_HEADER_MAP = {
-#     "machine_hex": PEFileHeader.machine,
-#     "time_date_stamp": PEFileHeader.time_date_stamp,
-#     "pointer_to_symbol_table": PEFileHeader.pointer_to_symbol_table,
-#     "number_of_symbols": PEFileHeader.number_of_symbols,
-#     "size_of_optional_header": PEFileHeader.size_of_optional_header,
-#     "characteristics": PEFileHeader.characteristics,
-#
-# }
-#
-# _PE_BINARY_OPTIONAL_HEADER_MAP = {
-#     "magic_hex": PEOptionalHeader.magic,
-#     "major_linker_version": PEOptionalHeader.major_linker_version,
-#     "minor_linker_version": PEOptionalHeader.minor_linker_version,
-#     "size_of_code": PEOptionalHeader.size_of_code,
-#     "size_of_initialized_data": PEOptionalHeader.size_of_initialized_data,
-#     "size_of_uninitialized_data": PEOptionalHeader.size_of_uninitialized_data,
-#     "address_of_entry_point": PEOptionalHeader.address_of_entry_point,
-#     "base_of_code": PEOptionalHeader.base_of_code,
-#     "base_of_data": PEOptionalHeader.base_of_data,
-#     "image_base": PEOptionalHeader.image_base,
-#     "section_alignment": PEOptionalHeader.section_alignment,
-#     "file_alignment": PEOptionalHeader.file_alignment,
-#     "major_os_version": PEOptionalHeader.major_os_version,
-#     "minor_os_version": PEOptionalHeader.minor_os_version,
-#     "major_image_version": PEOptionalHeader.major_image_version,
-#     "minor_image_version": PEOptionalHeader.minor_image_version,
-#     "major_subsystem_version": PEOptionalHeader.major_subsystem_version,
-#     "minor_subsystem_version": PEOptionalHeader.minor_subsystem_version,
-#     "win32_version_value_hex": PEOptionalHeader,
-#     "size_of_image": PEOptionalHeader.size_of_image,
-#     "size_of_headers":PEOptionalHeader.size_of_headers,
-#     "checksum_hex": PEOptionalHeader.checksum,
-#     "subsystem_hex": PEOptionalHeader.subsystem,
-#     "dll_characteristics_hex": PEOptionalHeader.dll_characteristics,
-#     "size_of_stack_reserve": PEOptionalHeader.size_of_stack_reserve,
-#     "size_of_stack_commit": PEOptionalHeader.size_of_stack_commit,
-#     "size_of_heap_reserve": PEOptionalHeader.size_of_heap_reserve,
-#     "size_of_heap_commit": PEOptionalHeader.size_of_heap_commit,
-#     "loader_flags_hex": PEOptionalHeader.loader_flags,
-#     "number_of_rva_and_sizes": PEOptionalHeader.number_of_rva_and_sizes
-# }
-
+    elif prop_name1 == "pdfid0" or prop_name1 == "pdfid1":
+        warn("Order may not be maintained for pdfids in %s", 514, id20)
```

Comment:
This seems like a convoluted way, which (I now understand) doesn't maintain the order. Why not just collect two values and then assemble them in the correct order?

Reply:
I guess I could do that - but its not trivial.  Remember, I'm traversing an expression tree - when I encounter one of the pdfids I don't know if I will see the other one. I'd have to wait until the traversal is over, and know to do some post processing.  Do you see an easier way?

Reply:
I forgot (it's been a few months since looking at this code closely ;-) ) that we generally just mutate objects in place. In that situation, this may be the best we can do. 

It would be interesting to see if a style where we construct objects directly rather than modifying them would work. Because patterns in STIX 1 and STIX 2 are so vastly different (and in some ways very difficult to reason about in STIX 1), that might not be possible.  I'm OK with the current approach for this PR.

Reply:
At the top level, it is determined what STIX 1.x/CybOX 2.x object is appropriate for the given STIX 2.0 object, and an empty object of that type is created and then filled in - so wouldn't exactly say it is "mutated"